### DISCUSSION TITLE
How to unnest a field

### DISCUSSION THREAD

**Discussion by wdfinch** [jOOQ/jOOQ#15054] [Q&A] [answered]
Hi there! I'm new to this library and had a question regarding `unnest`.  I'm storing a field with a`VARCHAR[]` datatype. I want to select and unnest that field so I can easily work with it. So something like the below. 

```
SELECT DISTINCT unnest(nestedField) AS unnestedField
FROM myTable
```

I can't seem to get this working with jooq. While there is an `unnest` function, I cannot use it in a select statement like the above.  Is this possible using jooq or do I need to write plain SQL? Thanks!

**Comment by wdfinch** [comment]
In jooq, this works:

```kotlin
ctx.select(field("DISTINCT unnest(nestedField) AS unnestedField"))
.from(TABLE.nestedField)
```

Can this be done in pure jooq?

**Comment by lukaseder** [comment, marked as answer]
Do any of these help?

- https://stackoverflow.com/a/35559962/521799
- https://stackoverflow.com/a/36576089/521799

**Reply by wdfinch** [reply]
Yes, thanks so much! This really helped me out.
